src/lib/storage.ts

```diff
@@ -0,0 +1,6 @@
+abstract class Storage {
```

Comment:
Storage shouldn't care about todos, There should be different type of keys and values we should be able to store. 

By referencing only todos we are limiting what it can do,